edge/services/helloSecretWorld/README.md

```diff
@@ -0,0 +1,122 @@
+
+  ```
+  hzn sm secret add --secretKey <secret-key> --secretDetail <new-secret-value> secret-name
+  ```
+
+A prompt will pop up in the terminal asking you to confirm that you want to overwrite the existing `secret-name` in the secrets manager. Enter `y` to confirm.
+
+6. After some time you should see the output of the service change to **\<your-node-id> says: Hello \<new-secret-value>!**, where `<new-secret-value>` is the new value of the secret as updated in the secrets manager in Step 17.
+
+7. Unregister your edge node (which will also stop the hello-secret service):
+
+  ```bash
+  hzn unregister -f
+  ```
+
+## <a id=secret-info></a> Secrets Manager Access Control and CLI
+
+### Access Control
+
+To access secrets using the Horizon secrets manager CLI, a user must provide their Exchange credentials through the environment variable `HZN_EXCHANGE_USER_AUTH` or can specify their credentials through flags passed to the command.
+
+Secrets are created within two different scopes: organization-wide secrets and user private secrets.
+
+Any authenticated user in the organization can list their organization-wide secrets, meaning they can see the names of secrets in their organization. However, only organization admins can read organization-wide secrets, meaning they can access key/value pairs of secrets using the CLI. Organization admins are also able to add, remove, and update organization-wide secrets.
+
+User private secrets are owned by individual users in the organization. These secrets will not be accessible by anyone other than the users that own them. User private secrets follow a naming rule; they take the form of `user/<user>/<name>`, where `<user>` is the Exchange ID of the user who owns that secret and `<name>` is the user-unique name of the secret. This rule is always used when referring to user private secrets in the secrets manager.
+
+### Secrets Manager Horizon CLI
+
+The Horizon CLI includes a subcommand `secretsmanager | sm` for organization administrators and users to access the resources in the secrets manager, with access restricted as described above.
+
+For organization administrators looking to manage the resources in their organization's secrets manager, the CLI has commands to `add`, `remove`, and `read` secrets in the secrets manager. These commands can also be used by users on their own private secrets. 
+
+For service deployers who need to specify secrets in their patterns and deployment policies, the `list` command allows them to list all the secrets in their organization, as well as check whether an organization-wide secret exists. Of course, this command can also be used by users on their own private secrets. To list a user's private secrets, the name `user/<user>` should be passed to `list`, where `<user>` is the owner of the secrets. Only the user who owns those secrets is able to do this.
+
```

Comment:
Can you add a section in here on secret names that covers:
- names with '/' characters - used to create hierarchical names
- restrictions on allowable characters
- re-iterates the /user/<user>/<secret-name> convention for user private secrets